kubernetes/helm_charts/datapipeline_jobs/templates/flink_job_deployment.yaml

```diff
@@ -204,3 +204,22 @@ spec:
         - name: flink-config-volume
           mountPath: /opt/flink/conf/log4j-console.properties
           subPath: log4j-console.properties
+
+---
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Release.Name }}-taskmanager-hpa
+  namespace: {{ .Values.namespace }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ .Release.Name }}-taskmanager
+  minReplicas: 1
+  maxReplicas: 2
+  metrics:
+    - type: External
+      external:
+        metricName: {{ .Release.Name }}_kafka_consumergroup_lag_sum
+        targetValue: "500000"
```

Comment:
@sowmya-dixit Could you please variablise this targetValue. Do we need to have thevariables/threshold value for each flink job or one value for all the jobs is enough?

Reply:
Updated HPA yaml to take configurations dynamically for each job